### DISCUSSION TITLE
Autodig Restock Pickaxes

### DISCUSSION THREAD

**Discussion by pepan-7** [ORelio/Minecraft-Console-Client#2352] [Q&A]
Has anyone made a script that replenishes pickaxes similarly to fishing rods in AutoFishishing? I've tried to port it from autofishing but brain too small for C#.  Also tried using MineCube.cs but the server I'm playing on kicks me when i enable it and i just need it to dig 1-2 blocks from a generator.

**Comment by milutinke** [comment]
Some solutions you could do without a script would be either having mending on pickaxe and shooting XP bottles from time to time in succession (rapid fire), or having the bot still in once place standing on a hopper and having a dispenser cycling pickaxes.
As for writing a script, you would have to have a container in which to store the pickaxe, then you would need to move to the location close to the container and then open it and take the item from it.

To move you could use: `MoveToLocation` method.
To open a chest you can use: `SendPlaceBlock` method and then catch the event with `OnInventoryOpen`.
Then use `GetInventories` to get inventories and search by inventory id you got from `OnInventoryOpen`, then use https://github.com/MCCTeam/Minecraft-Console-Client/blob/master/MinecraftClient/Inventory/ItemMovingHelper.cs methods to switch items.
